Examples/Algorithms/Digitization/include/ActsExamples/Digitization/ModuleClusters.hpp

```diff
@@ -0,0 +1,68 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Utilities/BinUtility.hpp"
+#include "ActsExamples/Digitization/MeasurementCreation.hpp"
+#include "ActsExamples/EventData/SimHit.hpp"
+
+#include <set>
+#include <unordered_map>
+#include <vector>
+
+namespace ActsExamples {
+class ModuleClusters {
+ public:
+  using simhit_t = SimHitContainer::size_type;
+
+  ModuleClusters(Acts::BinUtility segmentation,
+                 std::vector<Acts::BoundIndices> geoIndices, bool merge,
+                 double nsigma)
+      : m_segmentation(std::move(segmentation)),
+        m_geoIndices(std::move(geoIndices)),
+        m_merge(merge),
+        m_nsigma(nsigma) {}
+
+  void add(DigitizedParameters params, simhit_t simhit);
+  std::vector<std::pair<DigitizedParameters, std::set<simhit_t>>>
+  digitizedParameters();
+
+ private:
+  struct ModuleValue {
+    std::vector<Acts::BoundIndices> paramIndices = {};
+    std::vector<Acts::ActsScalar> paramValues = {};
+    std::vector<Acts::ActsScalar> paramVariances = {};
+    std::variant<Cluster, Cluster::Cell> value;
```

Comment:
Yes, in the end I think it's simpler to have a unified data structure for both cases. The parameter vectors are ~cheap to move around anyways since they can be std::move'd as required.